### PR TITLE
feat(issue-platform): Add payload_type to issue platform api

### LINKED ISSUE #58033: Support multiple message types for the issue platform API
The Issue Platform API should take in a `payload_type`. Currently, the only payload type is `occurrence` - this should also be the default if the `payload_type` isn't provided. 

Unsupported message types should be handled gracefully.

### PR DESCRIPTION
**Pull request by snigdhas**
Adds a new field, `payload_type`, to the issue platform API and adds the `occurrence` payload type. All calls to the API will default to the `occurrence` type if unspecified. 

Fixes https://github.com/getsentry/sentry/issues/58033

### CODE REVIEW

File: src/sentry/issues/producer.py [hunk 1]
```diff
@@ -26,25 +31,32 @@ def _get_occurrence_producer() -> KafkaProducer:
...
+                lookup_event_and_process_issue_occurrence,
+                process_event_and_issue_occurrence,
+            )
+
+            if event_data:
+                process_event_and_issue_occurrence(occurrence.to_dict(), event_data)
+            else:
+                lookup_event_and_process_issue_occurrence(occurrence.to_dict())
+            return
+        payload_data = cast(MutableMapping[str, Any], occurrence.to_dict())
```
Comment (JoshFerge) [inline src/sentry/issues/producer.py:56]: seems like we're missing test coverage for lines 56-60?

File: src/sentry/issues/producer.py [hunk 2]
```diff
@@ -26,25 +31,32 @@ def _get_occurrence_producer() -> KafkaProducer:
...
-        # If we're not running Kafka then we're just in dev. Skip producing to Kafka and just
-        # write to the issue platform directly
-        from sentry.issues.occurrence_consumer import (
-            lookup_event_and_process_issue_occurrence,
-            process_event_and_issue_occurrence,
-        )
-
+    if payload_type == PayloadType.OCCURRENCE:
+        if not occurrence:
+            raise ValueError("Occurrence must be provided")
```
Comment (JoshFerge) [inline src/sentry/issues/producer.py:40]: should we add a test case to cover this?

**Comment by JoshFerge** [comment]
what other payload types are there?

File: src/sentry/issues/producer.py (hunk 1 above)
Comment (JoshFerge) [inline src/sentry/issues/producer.py:56]: why do we have to cast here? should be avoided if possible

Comment (snigdhas) [reply src/sentry/issues/producer.py:56]: Not sure, we're just indenting the code here but can look into cleaning it up later.

**Comment by snigdhas** [comment]
Adding a new payload type in https://github.com/getsentry/sentry/issues/58034

**Comment by rjo100** [comment]
looks good for `crons`

**Review by wedamija — APPROVED** [review]

File: src/sentry/issues/producer.py [hunk 3]
```diff
@@ -26,25 +31,32 @@ def _get_occurrence_producer() -> KafkaProducer:
 
 
 def produce_occurrence_to_kafka(
-    occurrence: IssueOccurrence, event_data: Optional[Dict[str, Any]] = None
+    payload_type: PayloadType | None = PayloadType.OCCURRENCE,
```
[the remark this answers is not in the document]
Comment (snigdhas) [reply src/sentry/issues/producer.py:34]: will fix in a downstream PR, ty!

**Pull request merged**